src/library/scala/collection/Iterator.scala

```diff
@@ -1050,7 +1050,7 @@ object Iterator extends IterableFactory[Iterator] {
         tail = tail.tail
         merge()
         if (currentHasNextChecked) true
-        else if (current.hasNext) {
+        else if ((current ne null) && current.hasNext) {
```

Comment:
tiny nitpick: you can write `current != null`, it's more idiomatic, equally performant (same bytecode generated), and doesn't require the extra set of parens

Reply:
I see the file has a few other `ne null`s, though, so I guess cleaning that up is outside the scope of this PR. so never mind, I guess

Reply:
> equally performant (same bytecode generated)

Hm, I don't know that, thanks. Is it true only for null-check?